app/models/survey.rb

```diff
@@ -46,4 +47,8 @@ def questions
 
     Question.order("RANDOM()") + demographic_questions + customised_questions
   end
+
+  def self.master_survey
+    where(master: true).first
```

Comment:
I think we could use `find_by_master(true)` here which will return the first result. Also, we should probably put some kind of validation or contstraint that means you can only have one master survey at a time.

Reply:
Thanks - would you recommend using a before_save kind of action to achieve that constraint?

Reply:
Probably just a uniqueness validation https://stackoverflow.com/questions/19575336/rails-custom-validation-only-one-record-can-be-true

Reply:
Many thanks I am working to implement this.